### PR TITLE
Rake file and service to import placements from CSV

### PR DESCRIPTION
**Pull request by carlosmartinez**
[Trello](https://trello.com/c/W8r5cF1T/6451-import-student-record-placement-data-into-register)

We have a CSV file containing HESA IDs and URNs for placements. This commit adds a rake task and service to import placements on the basis of this file.

The file should be uploaded to the database using the Upload model (via the admin interface). The rake task can then be run with the ID of the upload.

### Guidance to review

On the review app, upload a CSV file with hesa_id and urn fields and check that the corresponding placements are created.

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: lib/tasks/import_hesa_placements.rake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+namespace :import_hesa_placements do
+  # Usage: rails import_hesa_placements:from_upload[upload_id]
```
Comment (darokel) [inline lib/tasks/import_hesa_placements.rake:4]: Could use `desc` to describe the task so it also appears when doing ` rails -T`

**Review by darokel — APPROVED** [review]
Left one comment but it's not a blocker 👍

File: app/services/placements/import_from_csv.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def initialize(upload_id:)
+      @upload = Upload.find(upload_id)
+    end
+
+    # Uploaded CSV file should contain hesa_id and urn headers only
+    def call
+      table = CSV.parse(upload.file.download, headers: true)
+      table.each do |row|
+        next unless (school = School.find_by(urn: row["urn"]))
+        next unless (trainee = Trainee.find_by(hesa_id: row["hesa_id"]))
```
Comment (d-a-v-e) [inline app/services/placements/import_from_csv.rb:16]: do we need to scope trainee (perhaps by academic year?) - I recall hesa_id not being guaranteed unique.

**Review by d-a-v-e — APPROVED** [review]
This looks great! glad we can finally get this in. Will it be first run on productiondata?

File: app/services/placements/import_from_csv.rb (hunk 2 above)
Comment (carlosmartinez) [reply app/services/placements/import_from_csv.rb:16]: Good point!

**Pull request merged**